File: public/api/esp32/capture.py
"""
ESP32 Capture API - Python Version
Chuyển đổi từ esp32/capture.php
"""
import sys
import os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..', '..', '..', 'src'))

from flask import request
import requests
from datetime import datetime
from helpers import json_out, ensure_upload_dir
from config.env import CONFIG

def capture():
    """Capture từ ESP32 và lưu file"""
    ip = request.args.get('ip', '')
    if not ip:
        return json_out({'ok': False, 'error': 'Missing ip'}, 422)
    
    try:
        url = f"http://{ip}/capture"
        response = requests.get(url, timeout=10)
        
        if response.status_code < 200 or response.status_code >= 300:
            return json_out({'ok': False, 'error': f"ESP32 returns HTTP {response.status_code}"}, 502)
        
        img_bytes = response.content
        
        day, day_path = ensure_upload_dir()
        timestamp = int(datetime.now().timestamp())
        filename = f"{timestamp}_{os.urandom(4).hex()}.jpg"
        
        with open(os.path.join(day_path, filename), 'wb') as f:
            f.write(img_bytes)
        
        base = CONFIG['app']['upload_base'].rstrip('/')
        public_url = f"{base}/{day}/{filename}"
        
        return json_out({'ok': True, 'url': public_url})
    except Exception as e:
        return json_out({'ok': False, 'error': f'Exception: {str(e)}'}, 500)

File: public/api/esp32/capture.php
<?php
// Project_Q/public/api/esp32/capture.php

// Tắt hiển thị lỗi PHP để không làm hỏng JSON response
ini_set('display_errors', '0');
error_reporting(E_ALL);

require __DIR__ . '/../../../src/bootstrap.php';

$ip = $_GET['ip'] ?? '';
if ($ip === '') json_out(['ok' => false, 'error' => 'Missing ip'], 422);

try {
    $url = "http://{$ip}/capture";

    $ch = curl_init($url);
    curl_setopt_array($ch, [
        CURLOPT_RETURNTRANSFER => true,
        CURLOPT_CONNECTTIMEOUT => 3,
        CURLOPT_TIMEOUT        => 10,
    ]);
    $bin  = curl_exec($ch);
    $err  = curl_error($ch);
    $code = curl_getinfo($ch, CURLINFO_HTTP_CODE) ?: 0;
    curl_close($ch);

    if ($bin === false || $code < 200 || $code >= 300) {
        json_out(['ok' => false, 'error' => $err ?: "ESP32 returns HTTP {$code}"], 502);
    }

    [$day, $dayPath] = ensure_upload_dir(); // đã có sẵn trong src/helpers.php
    $filename = time() . '_' . bin2hex(random_bytes(4)) . '.jpg';
    file_put_contents($dayPath . '/' . $filename, $bin);

    $base = rtrim($GLOBALS['cfg']['app']['upload_base'], '/'); // lấy từ config/env.php
    $publicUrl = "{$base}/{$day}/{$filename}";

    json_out(['ok' => true, 'url' => $publicUrl]);
} catch (Exception $e) {
    json_out(['ok' => false, 'error' => 'Exception: ' . $e->getMessage()], 500);
}
